stdlib/http/src/main/ballerina/src/http/cookie/cookieStore.bal

```diff
@@ -0,0 +1,343 @@
+        return true;
+    }
+    if (path.startsWith(cookie.path) && path[cookie.path.length()] == "/" ) {
+        return true;
+    }
+    return false;
+}
+
+// Returns true , if cookie expires attribute value is valid according to rfc-6265. - https://tools.ietf.org/html/rfc6265#section-5.1.1
+function isExpiresAttributeValid(Cookie cookie) returns boolean {
+    if (cookie.expires != "") {
+        time:Time|error t1 = time:parse(cookie.expires.substring(0, cookie.expires.length() - 4), "E, dd MMM yyyy HH:mm:ss");
+        if (t1 is time:Time) {
+            int year = time:getYear(t1);
+            if (year <= 69 && year >= 0) {
+                time:Time tmAdd = time:addDuration(t1, 2000, 0, 0, 0, 0, 0, 0);
+                string|error timeString = time:format(tmAdd, "E, dd MMM yyyy HH:mm:ss");
+                if (timeString is string) {
+                    cookie.expires = timeString + " GMT";
+                    return true;
+                } else {
+                    return false;
+                }
+            }
+            return true;
+        } else {
+            return false;
+        }
+    }
+    return true;
+}
+
+// Adds a persistent cookie to the cookie store according to the rules in rfc-6265. - https://tools.ietf.org/html/rfc6265#section-5.3 , https://tools.ietf.org/html/rfc6265#section-4.1.2
+function addPersistentCookie(Cookie? identicalCookie, Cookie cookie, string url, CookieStore cookieStore) {
+    if (identicalCookie is Cookie) {
+        if (isExpired(cookie)) {
+             boolean isRemoved = cookieStore.removeCookie(identicalCookie.name, identicalCookie.domain, identicalCookie.path);
+        } else {
+            // Removes old cookie and add new persistent cookie.(replace similar cookies)
+            if ((identicalCookie.httpOnly == true && url.startsWith("http")) || identicalCookie.httpOnly == false) {
+                boolean isRemoved = cookieStore.removeCookie(identicalCookie.name, identicalCookie.domain, identicalCookie.path);
+                cookie.creationTime = identicalCookie.creationTime;
+                cookie.lastAccessedTime = time:currentTime();
+                // TODO:insert into the database.
+            }
+        }
+    } else {
+        // Adds if cookie is not expired.
+        if (!isExpired(cookie)) {
+            cookie.creationTime = time:currentTime();
+            cookie.lastAccessedTime = time:currentTime();
+            // TODO:insert into the database.
+        }
+    }
+}
+
+// Returns true , if cookie is expired according to the rules in rfc-6265. - https://tools.ietf.org/html/rfc6265#section-4.1.2.2
+function isExpired(Cookie cookie) returns boolean {
+    if (cookie.maxAge > 0) {
+        time:Time exptime = time:addDuration(cookie.creationTime, 0, 0, 0, 0, 0, cookie.maxAge, 0);
+        time:Time curTime = time:currentTime();
+        if (exptime.time < curTime.time) {
+            return true;
+        } else {
```

Comment:
Can we just return false here, without having it inside an else block? Line 320 also?